fixieai/cli/validators.py

```diff
@@ -0,0 +1,17 @@
+import re
+
+AGENT_ID_PATTERN = re.compile(r"^\w+$")
+URL_PATTERN = re.compile(
+    r"^https?:\\/\\/(?:www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b"
+    r"(?:[-a-zA-Z0-9()@:%_\\+.~#?&\\/=]*)$"
+)
+
+
+def validate_agent_id(agent_id: str):
+    if not AGENT_ID_PATTERN.match(agent_id):
+        raise ValueError("agent_id must be alpha numerics and underscores only.")
+
+
+def validate_url(url: str):
```

Comment:
If you follow my recommendation above to use Click's built-in prompting for options, you can hook up the validators to Click trivially as well:

https://click.palletsprojects.com/en/8.1.x/options/#callbacks-for-validation